packages/@ember/-internals/glimmer/lib/templates/link-to.ts

```diff
@@ -1,30 +1,21 @@
-import { precompileTemplate } from '@ember/template-compilation';
-import { on } from '@ember/modifier';
+import { hbs } from '@lifeart/gxt';
 
-export default precompileTemplate(
-  `<a
-  {{!-- for compatibility --}}
-  id={{this.id}}
-  class={{this.class}}
+export default function LinkToTemplate() {
+  return hbs`<a
+    {{!-- for compatibility --}}
+    id={{this.id}}
+    class={{this.class}}
 
-  {{!-- deprecated attribute bindings --}}
-  role={{this.role}}
-  title={{this.title}}
-  rel={{this.rel}}
-  tabindex={{this.tabindex}}
-  target={{this.target}}
+    {{!-- deprecated attribute bindings --}}
+    role={{this.role}}
+    title={{this.title}}
+    rel={{this.rel}}
+    tabindex={{this.tabindex}}
+    target={{this.target}}
 
-  ...attributes
+    ...attributes
 
-  href={{this.href}}
+    href={{this.href}}
 
-  {{on 'click' this.click}}
->{{yield}}</a>`,
-  {
-    moduleName: 'packages/@ember/-internals/glimmer/lib/templates/link-to.hbs',
-    strictMode: true,
-    scope() {
-      return { on };
-    },
-  }
-);
+    {{on 'click' this.click}} >{{yield}}</a>`;
+}
```

Comment:
What is the motivation for needing to rewrite these? 

You could reimplement the meaning of `precompileTemplate` to be whatever you need it to be, and then you wouldn't need to touch this code and you would have a much greater change of actually achieving upgrade compatibility.

Reply:
@ef4 this is just for experimentation purposes, I agree - we could keep same api as-is here (compiler need to be extended a bit)